api/v1alpha1/oidc_types.go

```diff
@@ -143,6 +150,19 @@ type OIDCProvider struct {
 	TokenEndpoint *string `json:"tokenEndpoint,omitempty"`
 }
 
+// OIDCDenyRedirectMatcher defines the matcher to deny redirect to the OIDC Provider.
+// +notImplementedHide
+type OIDCDenyRedirectMatcher struct {
+	// Specifies the name of the header in the request.
+	// The pseudo-headers ``:path`` and ``:method`` can be used to match the request path and method, respectively
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name"`
+	// If specified, header match will be performed based on the string match of the header value.
+	// Specifies how the header match will be performed to route the request.
+	// +optional
+	StringMatch StringMatch `json:"stringMatch,omitempty"`
```

Comment:
It also worth to mention in the comment that the matcher will only check the presence of the header if the `stringMatcher` is not specified.
```suggestion
	StringMatch *StringMatch `json:"stringMatch,omitempty"`
```

Reply:
I think its not optional. If a denyRedirectMatcher is specified it will ned to have a StringMatch of some type.
The whole denyRedirectMatcher is optional.

I will remove +optional from StringMatch